File: Euler5/Euler5.cs
/* 2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.

What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?

We are going to achieve this by testing the divisability of our number after increasing it. We always have to increase
it for atleast the highest Divisor we have.
 */
static int findSmallestByAllDivisable(int n){
    bool allDivisable = false;
    int num = 0;
    while(!allDivisable){
        num+=20;
        allDivisable = false;
        for(int i = n; i>0; i--){
            allDivisable = (num%i)==0;
            if(!allDivisable){break;}
        }
    }
    return num;
}
Console.WriteLine(findSmallestByAllDivisable(20));
//correct solution 232792560

File: Euler5/Euler5.rs
fn main() {
    let mut x = 20;
    while !divisible_till_20(&x) {
        x += 20;
    }
    println!("We have a number: {}", x);
}

fn divisible_till_20(x: &u64) -> bool {
    let mut is_divisible = true;
    let mut i = 1;
    while is_divisible && i <= 20 {
        is_divisible = x % i == 0;
        i += 1;
    }
    is_divisible
}
